src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,64 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private int size;
+    private Node<T> head;
```

Comment:
better call it tail, IMHO

Reply:
up to you

Reply:
Okay, considering the code below, I would suggest you fix all @Max-Pochepets  comments. Push method Should not iterate through the whole stack to find the last one. You should keep your current node at the last position of your stack.